solidity/random-beacon/README.adoc

```diff
@@ -177,103 +166,122 @@ When the hard timeout is reached, anyone can notify about this fact by calling
-operator inactivity claim. If the required threshold of group members signed
+Group members can agree to punish members who are permanently inactive and issue
+an operator inactivity claim. If the required threshold of group members signed
 the operator inactivity claim, they can submit it to
-`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce)`
+`RandomBeacon.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce, int32[] calldata groupMembers)`
 function and have the group members who are inactive excluded from
 the sortition pool rewards for a governable time period.
 
 This approach is theoretically susceptible to group members colluding together
 but because a reasonably high number of operators is needed to sign a claim and
-operators signing the claim other than the submitter receive nothing in return,
+operators signing the claim receive nothing in return,
 we consider this approach safe and good enough. An important advantage of this
 approach is that honest players can decide off-chain when it makes sense to
 submit an operator inactivity claim and mark someone as ineligible for rewards.
 For example, marking an operator ineligible for rewards for the next two weeks
 have a higher impact than prolonging reward ineligibility for 10 minutes for an
 operator that was already marked as ineligible for rewards. This approach does
 not increase the gas cost of a happy path and leaves some freedom to group
-members. They may mark as ineligible operators who turned off their nodes,
+members. They can mark as ineligible operators who turned off their nodes,
 operators whose nodes never participate in signing because they are
 misconfigured, or operators who notoriously miss their turn in submitting relay
-entry.
+entries.
+
+=== Rewards
+
+T rewards are allocated to all operators registered in the beacon sortition
+pool, excluding operators who were marked as ineligible for rewards as a result
+of <<inactivity, inactivity>>. Rewards are allocated proportionally to the
+operator's weight in the pool. 
+
+[[transaction-incentives]]
+=== Transaction Incentives
+
+There are two types of transactions: <<operator-only,Operator-Only>>,
+<<public-knowledge,Public-Knowledge>>, and <<punishment,Punishment>>.
+
+[[operator-only]]
+==== Operator-Only
+Operator-Only transactions are where only the operators have access to the
+information required to assemble the transaction with the right input
+parameters.
+
+In order to avoid all operators racing to submit the transaction at the same
+time, we have an off-chain informal agreement to submit based on the operator's
+position in the group (can use the hash of the group's pubkey).
+
+If the designated operator does not submit their transaction before a timeout
+expires, the duty moves to the next operator and the group can sign a
+transaction to mark that operator as inactive. Since there is no slashing
```

Comment:
Link `inactive` to `[[inactivity]]`?

Reply:
1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322